src/services/implementation/ClientServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package services.implementation;
+
+import entities.Client;
+import enums.StatusEnum;
+import exceptions.services.EntityNotFoundException;
+import exceptions.services.IdNullPointerException;
+import exceptions.services.NullEntityException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import services.interfaces.IClientInterface;
+import util.ClientServiceUtility;
+
+public class ClientServiceImpl implements IClientInterface {
+  private List<Client> clients;
+  private Long currentId;
+
+  public ClientServiceImpl() {
+    this.clients = new ArrayList<>();
+    this.currentId = 0L;
+  }
+
+  @Override
+  public Optional<Client> insert(Client client) {
+    client.setId(this.currentId);
+    client.setCode(ClientServiceUtility.createCode(this.currentId));
+    this.currentId++;
```

Comment:
Don't use ++, please be more explicit for sake of readability

Reply:
Change to `x = x + 1`

Reply:
solved